Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -77,51 +63,75 @@ public struct FFTView: View {
     private var paddingFraction: CGFloat
     private var includeCaps: Bool
     private var node: Node
-    private var numberOfBars: Int
-    private var minAmplitude: Double
-    private var maxAmplitude: Double
+    private var barCount: Int
+    private var fftValidBinCount: FFTValidBinCount?
+    private var minAmplitude: Float
+    private var maxAmplitude: Float
+    private let defaultBarCount: Int = 64
+    private let maxBarCount: Int = 128
 
     public init(_ node: Node,
                 linearGradient: LinearGradient = LinearGradient(gradient: Gradient(colors: [.red, .yellow, .green]),
                                                                 startPoint: .top,
                                                                 endPoint: .center),
                 paddingFraction: CGFloat = 0.2,
                 includeCaps: Bool = true,
-                numberOfBars: Int = 50,
-                maxAmplitude: Double = -10.0,
-                minAmplitude: Double = -150.0)
+                validBinCount: FFTValidBinCount? = nil,
+                barCount: Int? = nil,
+                maxAmplitude: Float = -10.0,
+                minAmplitude: Float = -150.0)
     {
         self.node = node
         self.linearGradient = linearGradient
         self.paddingFraction = paddingFraction
         self.includeCaps = includeCaps
-        self.numberOfBars = numberOfBars
         self.maxAmplitude = maxAmplitude
         self.minAmplitude = minAmplitude
+        self.fftValidBinCount = validBinCount
 
         if maxAmplitude < minAmplitude {
             fatalError("Maximum amplitude cannot be less than minimum amplitude")
         }
         if minAmplitude > 0.0 || maxAmplitude > 0.0 {
             fatalError("Amplitude values must be less than zero")
         }
+
+        if let requestedBarCount = barCount {
+            self.barCount = requestedBarCount
+        } else {
+            if let fftBinCount = fftValidBinCount {
+                if Int(fftBinCount.rawValue) > maxBarCount - 1 {
+                    self.barCount = maxBarCount
+                } else {
+                    self.barCount = Int(fftBinCount.rawValue)
+                }
+            } else {
+                self.barCount = defaultBarCount
```

Comment:
instead of this third option, can we just crash or something? i feel like if you haven't provided barCount or a valid binCount then we should just give up.

Reply:
There may be a better way to go about the part you are referring to there, but I really think it's best to have some default FFTView appearance.